tests/system_tests_split_path.py

```diff
@@ -0,0 +1,235 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+"""
+Verifies that the router detects when the inter-router connection group is
+split across routers
+"""
+
+import os
+import selectors
+import socket
+from threading import Event, Thread
+from system_test import TestCase, main_module, Qdrouterd, unittest
+from system_test import TIMEOUT, Logger
+
+
+class MyLoadBalancer:
```

Comment:
Can we try using haproxy to do tcp load balancing (round robin) instead of writing our own load balancer ?

https://www.haproxy.com/documentation/haproxy-configuration-tutorials/load-balancing/tcp/

Reply:
Nginx also allows TCP loadbalancing but unfortunately that feature is only available to `Nginx Plus` users.
https://docs.nginx.com/nginx/admin-guide/load-balancer/tcp-udp-load-balancer/

Reply:
I don't have any other test cases that require haproxy and the mock load balancer gives the patch full coverage. Can haproxy be added to the CI infrastructure as a follow on to this patch?